src/core/expressioncontextutils.cpp

```diff
@@ -20,10 +20,21 @@
 #include "qgsgeometry.h"
 #include <QtPositioning/QGeoPositionInfoSource>
 
-QgsExpressionContextScope *ExpressionContextUtils::positionScope( const GnssPositionInformation &positionInformation )
+
+void addVar( QgsExpressionContextScope *scope, const QString &name, const QVariant &value, bool positionLocked, const QVariant &defaultValue = QVariant() )
```

Comment:
what about declaring the function _within_ ExpressionContextUtils::positionScope? The gnss,position variable prefixes below make this function only usable within that positionScope.

Alternative, rename to addPositionVariable?

Reply:
I thought it's easier to read without the lambda syntax, so let's go for `addPositionVariable`

Reply:
```suggestion
void addPositionVariable( QgsExpressionContextScope *scope, const QString &name, const QVariant &value, bool positionLocked, const QVariant &defaultValue = QVariant() )
```